src/marine/avhrr_radiance2ioda.py

```diff
@@ -0,0 +1,286 @@
+    )
+
+    required = parser.add_argument_group(title='required arguments')
+    required.add_argument(
+        '-i', '--input',
+        help="path of patmosx avhrr observation input file(s)",
+        type=str, nargs='+', required=True)
+    required.add_argument(
+        '-o', '--output',
+        help="path of IODA output file",
+        type=str, required=True)
+    required.add_argument(
+        '-d', '--date',
+        metavar="YYYYMMDDHH",
+        help="base date for the center of the window",
+        type=str, required=True)
+
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument(
+        '-t', '--thin',
+        help="percentage of random thinning, from 0.0 to 1.0. Zero indicates"
+             " no thinning is performed. (default: %(default)s)",
+        type=float, default=0.0)
+    optional.add_argument(
+        '--threads',
+        help='multiple threads can be used to load input files in parallel.'
+             ' (default: %(default)s)',
+        type=int, default=1)
+
+    args = parser.parse_args()
+    args.date = datetime.strptime(args.date, '%Y%m%d%H')
+
+    # Setup the configuration that is passed to each worker process
+    # Note: Pool.map creates separate processes, and can only take iterable
+    # objects. Rather than using global variables, embed them into
+    # the iterable object together with argument array passed in (args.input)
+    global_config = {}
+    global_config['date'] = args.date
+    global_config['thin'] = args.thin
+    global_config['oval_name'] = iconv.OvalName()
+    global_config['oerr_name'] = iconv.OerrName()
+    global_config['opqc_name'] = iconv.OqcName()
+
+    pool_inputs = [(i, global_config) for i in args.input]
+
+    # read / process files in parallel
+    pool = Pool(args.threads)
+    obs = pool.map(read_input, pool_inputs)
+
+    # concatenate the data from the files
+    obs_data, GlobalAttrs = obs[0]
+
+    for i in range(1, len(obs)):
+        for k in obs_data:
+            axis = len(obs[i][0][k].shape)-1
+            obs_data[k] = np.concatenate(
+                (obs_data[k], obs[i][0][k]), axis=axis)
+
+    # prepare global attributes we want to output in the file,
+    # in addition to the ones already loaded in from the input file
+    GlobalAttrs['date_time_string'] = args.date.strftime("%Y-%m-%dT%H:%M:%SZ")
+    GlobalAttrs['thinning'] = args.thin
+    GlobalAttrs['converter'] = os.path.basename(__file__)
+
+    # determine which variables we are going to output
+    selected_names = []
+    selected_names.append(output_var_names[0])
+
+    # pass parameters to the IODA writer
+    # (needed because we are bypassing ExtractObsData within BuildNetcdf)
+    VarDims = {
+        'brightness_temperature': ['nlocs', 'nchans']
+    }
+
+    nchans = len(chan_number)
+    nlocs = len(obs_data[('longitude', 'MetaData')])
+    DimDict = {
+        'nlocs': nlocs,
+        'nchans': nchans
```

Comment:
This is resulting in the nchans variable being set to all zeros, where it should be set to the list of channel numbers. The API will accept a list of the channel numbers which will then assign those channel numbers to the nchans variable. You should be able to fix this by changing this line to:

`'nchans': list(chan_number)`

Reply:
I see. Thanks